src/Microsoft.Dnx.Tooling/Publish/PublishManager.cs

```diff
@@ -176,17 +181,43 @@ public bool Publish()
                     frameworkContexts[frameworkName] = CreateDependencyContext(project, frameworkName);
                 }
             }
-
+         
             // If there is no target framework filter specified with '--runtime',
             // the published output targets all frameworks specified in project.json
             if (!_options.Runtimes.Any())
             {
-                foreach (var frameworkInfo in project.GetTargetFrameworks())
+                IEnumerable<FrameworkName> frameworksToPublish;
+
+                if (!_options.TargetFrameworks.Any())
+                {
+                    frameworksToPublish = project
+                            .GetTargetFrameworks()
+                            .Select(fx => fx.FrameworkName);
+                }
+                else
+                {
+                    string frameworkSelectionError;
+                    frameworksToPublish = FrameworkSelectionHelper.SelectFrameworks(project,
+                                                                       _options.TargetFrameworks,
+                                                                       _applicationEnvironment.RuntimeFramework,
+                                                                       out frameworkSelectionError)?.ToList();
+                    if (frameworksToPublish == null)
+                    {
+                        _options.Reports.WriteError(frameworkSelectionError);
+                        return false;
+                    }
+
+                    foreach(var framework in frameworksToPublish)
+                    {
+                        root.Frameworks.Add(framework, null);
```

Comment:
We do this for all frameworks selected but we did not do this when the active runtime is being used (in the branch above) - why?